commands/operator-sdk/cmd/scorecard/resource_handler.go

```
@@ -130,6 +133,35 @@ func createFromYAMLFile(yamlPath string) error {
 			}
 		}
 		addResourceCleanup(obj, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		if obj.GetKind() == "Deployment" {
+			dep := &appsv1.Deployment{}
+			err = runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: viper.GetString(NamespaceOpt), Name: deploymentName}, dep)
+			if err != nil {
+				return fmt.Errorf("failed to get newly created deployment: %v", err)
+			}
+			set := labels.Set(dep.Spec.Selector.MatchLabels)
+			// in some cases, the pod from the old deployment will be picked up instead of the new one
+			err = wait.PollImmediate(time.Second*1, time.Second*60, func() (bool, error) {
+				pods := &v1.PodList{}
+				err = runtimeClient.List(context.TODO(), &client.ListOptions{LabelSelector: set.AsSelector()}, pods)
+				if err != nil {
+					return false, fmt.Errorf("failed to get list of pods in deployment: %v", err)
+				}
+				// make sure the pods exist
+				// there should only be 1 pod per deployment
```

same as before, we should throw an error if there are multiple pods showing up